warehouse/warehouse.go

```diff
@@ -1493,31 +1476,60 @@ func clearTriggeredUpload(wh model.Warehouse) {
 }
 
 func healthHandler(w http.ResponseWriter, _ *http.Request) {
-	dbService := ""
-	pgNotifierService := ""
+	var dbService, pgNotifierService string
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
 	if runningMode != DegradedMode {
-		if !CheckPGHealth(notifier.GetDBHandle()) {
+		if !CheckPGHealth(ctx, notifier.GetDBHandle()) {
 			http.Error(w, "Cannot connect to pgNotifierService", http.StatusInternalServerError)
 			return
 		}
 		pgNotifierService = "UP"
 	}
 
 	if isMaster() {
-		if !CheckPGHealth(dbHandle) {
+		if !CheckPGHealth(ctx, dbHandle) {
 			http.Error(w, "Cannot connect to dbService", http.StatusInternalServerError)
 			return
 		}
 		dbService = "UP"
 	}
 
-	healthVal := fmt.Sprintf(
-		`{"server":"UP","db":%q,"pgNotifier":%q,"acceptingEvents":"TRUE","warehouseMode":%q,"goroutines":"%d"}`,
-		dbService, pgNotifierService, strings.ToUpper(warehouseMode), runtime.NumGoroutine(),
+	healthVal := fmt.Sprintf(`
+		{
+			"server": "UP",
+			"db": %q,
+			"pgNotifier": %q,
+			"acceptingEvents": "TRUE",
+			"warehouseMode": %q,
+			"goroutines": "%d"
+		}
+	`,
+		dbService,
+		pgNotifierService,
+		strings.ToUpper(warehouseMode),
+		runtime.NumGoroutine(),
 	)
+
 	_, _ = w.Write([]byte(healthVal))
 }
 
+func CheckPGHealth(ctx context.Context, db *sql.DB) bool {
+	if db == nil {
+		return false
+	}
+
+	rows, err := db.QueryContext(ctx, `SELECT 'Rudder Warehouse DB Health Check'::text as message;`)
```

Comment:
Why did we remove the `rows.Err()` check

Reply:
Since we are not doing scan, no need to check for Err() as well?

Reply:
Although, since we are enforced to adding `rows.Err()` due to the upgrade of `golangci` version, ignoring error for `rows.Err()`

Reply:
Are you sure this is good enough? You're not even checking the result of the query.

Anyway, I like the idea implemented by this library [here](https://github.com/dimiro1/health/blob/master/db/checker.go#L28). Maybe we could do the same, `SELECT VERSION()` and then check that we actually got something valid out of it.

Reply:
We can do `QueryRowContext` if we care about reading the response or just `db.ExecContext` if we ignore results
```suggestion
	err := db.QueryRowContext(ctx, "SELECT  'Rudder Warehouse DB Health Check'::text as message;").Scan(&msg)

```

Reply:
Should I just compare the msg with `Rudder Warehouse DB Health Check`?

Reply:
Comparing msg now.